src/ripple/core/sql_backend/SQLInterface_sqlite.cpp

```diff
@@ -0,0 +1,3747 @@
+        std::uint32_t offset,
+        int limit,
+        int limit_used,
+        bool binary,
+        bool count,
+        bool bUnlimited);
+};
+
+static bool
+exists(SQLDatabase& db)
+{
+    if (db)
+    {
+        SQLDatabase_sqlite* sdb = dynamic_cast<SQLDatabase_sqlite*>(&*db);
+        if (sdb)
+            switch (sdb->db_.index())
+            {
+                case 1:
+                    return true;
+                case 2:
+                    return true;
+            }
+    }
+    return false;
+}
+
+static soci::session&
+getSession(SQLDatabase& db)
+{
+    if (db)
+    {
+        SQLDatabase_sqlite* sdb = dynamic_cast<SQLDatabase_sqlite*>(&*db);
+        if (sdb)
+            switch (sdb->db_.index())
+            {
+                case 1:
+                    return std::get<1>(sdb->db_);
+                case 2:
+                    return std::get<2>(sdb->db_).getSession();
+            }
+    }
+    Throw<std::runtime_error>("SQL Database is not initialized.");
+}
+
+static LockedSociSession
+checkoutDb(SQLDatabase& db)
+{
+    if (db)
+    {
+        SQLDatabase_sqlite* sdb = dynamic_cast<SQLDatabase_sqlite*>(&*db);
+        if (sdb)
+            switch (sdb->db_.index())
+            {
+                case 2:
+                    return std::get<2>(sdb->db_).checkoutDb();
+            }
+    }
+    Throw<std::runtime_error>("SQL Database is not initialized.");
+}
+
+static LockedSociSession
+checkoutDb(SQLDatabase_* db)
+{
+    if (db)
+    {
+        SQLDatabase_sqlite* sdb = dynamic_cast<SQLDatabase_sqlite*>(db);
+        if (sdb)
+            switch (sdb->db_.index())
+            {
+                case 2:
+                    return std::get<2>(sdb->db_).checkoutDb();
+            }
+    }
+    Throw<std::runtime_error>("SQL Database is not initialized.");
+}
+
+static void
+setupCheckpointing(SQLDatabase& db, JobQueue* q, Logs& l)
+{
+    if (db)
+    {
+        SQLDatabase_sqlite* sdb = dynamic_cast<SQLDatabase_sqlite*>(&*db);
+        if (sdb)
+            switch (sdb->db_.index())
+            {
+                case 2:
+                    std::get<2>(sdb->db_).setupCheckpointing(q, l);
+                    return;
+            }
+    }
+    Throw<std::runtime_error>("SQL Database is not initialized.");
+}
+
+std::string
+SQLInterface_sqlite::getDBName(SQLInterface::DatabaseType type)
+{
+    switch (type)
+    {
+        case LEDGER:
+        case LEDGER_SHARD:
+            return LgrDBName;
+        case TRANSACTION:
+        case TRANSACTION_SHARD:
+            return TxDBName;
+        case WALLET:
+            return WalletDBName;
+        case ACQUIRE_SHARD:
+            return AcquireShardDBName;
+        case ARCHIVE:
+        case STATE:
+        case DOWNLOAD:
+        case VACUUM:
+            return stateDBName;
+        case PEER_FINDER:
+            return "peerfinder";
+    }
+    return "Unknown";
+}
+
+std::tuple<bool, SQLDatabase, SQLDatabase>
+SQLInterface_sqlite::makeLedgerDBs(
```

Comment:
This function makes the ledger database and the transaction database. I would change the name